### PR TITLE
Add OpenTracing and OpenTelemetry sample

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

Adds OpenTracing and OpenTelemetry sample using Jaeger

### CODE REVIEW

**Review by sw-yx — APPROVED** [review]
can you do a short video sometime explaining this sample? i think would really help people try it out. i can be there if it would help, or you can do solo

**Comment by tsurdilo** [comment]
@Spikhalskiy updated to also add OpenTelemetry support

File: README.md [hunk 1]
```diff
@@ -146,6 +147,10 @@ All tests are available under [src/test/java](https://github.com/temporalio/samp
 
 - [**Set up SDK metrics**](https://github.com/temporalio/samples-java/tree/master/src/main/java/io/temporal/samples/metrics): Demonstrates how to set up and scrape SDK metrics.
 
+### OpenTracing Support
```
Comment (Spikhalskiy) [inline README.md:150]: Tracing?

File: README.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ This repository contains sample Workflow applications that demonstrate various c
     - [Scenario-based samples](#scenario-based-samples)
     - [API demonstrations](#api-demonstrations)
     - [SDK metrics](#sdk-metrics)
+    - [OpenTracing support](#opentracing-support)
```
Comment (Spikhalskiy) [inline README.md:20]: Tracing?

File: build.gradle [hunk 3]
```diff
@@ -45,6 +46,12 @@ dependencies {
     implementation group: 'io.micrometer', name: 'micrometer-registry-prometheus', version: '1.8.2'
     implementation group: 'net.thisptr', name: 'jackson-jq', version: '1.0.0-preview.20210928'
     implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.12.5'
+    implementation 'io.jaegertracing:jaeger-client:1.8.0'
+    implementation 'io.opentelemetry:opentelemetry-sdk:1.11.0'
```
Comment (Spikhalskiy) [inline build.gradle:50]: May you consider using opentelemetry-bom the same way we use jackson-bom? We should promote this style.

**Review by Spikhalskiy — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (tsurdilo) [reply README.md]: updated

File: README.md (hunk 2 above)
Comment (tsurdilo) [reply README.md]: updated

File: build.gradle (hunk 3 above)
Comment (tsurdilo) [reply build.gradle]: updated

**Pull request merged**